### PR TITLE
[eclipse/xtext#2127] Update to EMF 2022-03 (2.29.0)

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#2127] Update to EMF 2022-03 (2.29.0)

### CODE REVIEW

**Comment by cdietrich** [comment]
i have the feeling we dont have a workflow to regenerate all genmodels. e.g. 

```
./org.eclipse.xtext.tests/src/org/eclipse/xtext/util/GenericsTest.genmodel:    copyrightFields="false" runtimeVersion="2.29">
./org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/ecore/Genmodel.genmodel:    editorPluginID="org.eclipse.xtext.tests.editor" runtimeVersion="2.29" usedGenPackages="platform:/resource/org.eclipse.emf.ecore/model/Ecore.genmodel#//ecore">
./org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/ecore/GenModelAccessTest.genmodel:    copyrightFields="false" runtimeVersion="2.29">
./org.eclipse.xtext.tests/src/org/eclipse/xtext/linking/lazy/LazyLinkingTestLanguage.genmodel:    complianceLevel="8.0" copyrightFields="false" runtimeVersion="2.29" usedGenPackages="../../../../../../../org.eclipse.emf.ecore/model/Ecore.genmodel#//ecore">
./org.eclipse.xtext.xtext.generator/src-gen/org/eclipse/xtext/xtext/generator/parser/antlr/splitting/SimpleExpressions.genmodel:    editorPluginID="org.eclipse.xtext.xtext.generator.editor" runtimeVersion="2.29">
```

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ecore/EMFGeneratorFragment2.xtend [hunk 1]
```diff
@@ -195,6 +195,8 @@ class EMFGeneratorFragment2 extends AbstractXtextGeneratorFragment {
 	 *   <li>"JDK60"</li>
 	 *   <li>"JDK70"</li>
 	 *   <li>"JDK80"</li>
+	 *   <li>"JDK110"</li>
+	 *   <li>"JDK170"</li>
 	 * </ul>
 	 * The default level is "JDK80".
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ecore/EMFGeneratorFragment2.xtend:201]: This should be changed to JDK110

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ecore/EMFGeneratorFragment2.xtend [hunk 2]
```diff
@@ -172,14 +172,14 @@ class EMFGeneratorFragment2 extends AbstractXtextGeneratorFragment {
 	@Accessors(PUBLIC_SETTER)
 	boolean suppressLoadInitialization = false
 
-	/* Default to 2.20 if available, otherwise #get will return null */
-	GenRuntimeVersion emfRuntimeVersion = GenRuntimeVersion.get(GenRuntimeVersion.EMF220_VALUE)
+	/* Default to 2.29 if available, otherwise #get will return null */
+	GenRuntimeVersion emfRuntimeVersion = GenRuntimeVersion.get(GenRuntimeVersion.EMF229_VALUE)
 	GenJDKLevel jdkLevel = GenJDKLevel.JDK80_LITERAL
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ecore/EMFGeneratorFragment2.xtend:177]: Should be changed to JDK110_LITERAL

**Review by szarnekow — APPROVED** [review]

**Comment by szarnekow** [comment]
> i have the feeling we dont have a workflow to regenerate all genmodels.

Certainly true. Unfortunately.

**Pull request merged**